pkg/manager/manager.go

```diff
@@ -262,21 +252,16 @@ func (m *SupportBundleManager) collectNodeBundles() error {
 		return err
 	}
 
-	logrus.Infof("wating node bundles, (timeout: %s)", m.WaitTimeout)
-	select {
-	case <-m.ch:
-		logrus.Info("all node bundles are received.")
+	<-m.ch
+	logrus.Info("all node bundles are received.")
 
-		// Clean up when everything is fine. If something went wrong, keep ds for debugging.
-		// The ds will be garbage-collected when manager pod is gone.
-		err := agents.Cleanup()
-		if err != nil {
-			return errors.Wrap(err, "fail to cleanup agent daemonset")
-		}
-		return nil
-	case <-time.After(m.WaitTimeout):
-		return fmt.Errorf("fail to wait node bundles. missing: %+v", m.expectedNodes)
```

Comment:
@shangma  I added the timeout here because I thought a support bundle still has its value even not all node bundles are collected (and the k8s YAML/logs are already collected). In production clusters, it's common that some nodes are dead or can't respond in time.